tutorials/misc/running_code_in_the_editor.rst

```diff
@@ -0,0 +1,93 @@
+- If you have jumppads with varying jump heights, you can draw the maximum jump height a player would reach if it jumped on one, also making level design easier.
+- If your player doesn't use a sprite, but draws itself using code, you can make that drawing code execute in the editor to see your player.
+
+How to use it
+-------------
+
+To turn a script into a tool, add ``tool`` keyword at the top of your code, right under the ``extends ...`` line.
+
+To check if you are currently in the editor, use: ``Engine.editor_hint``.
+
+For example, if you want to execute some code only in editor, use:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    if Engine.editor_hint:
+        # code to execute when in editor
+
+On the other hand, if you want to execute code only in game, simply negate previous expression:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    if not Engine.editor_hint:
+        # code to execute when in game
+
+Pieces of code that are not having any of the 2 conditions above will run both in editor and in game.
+
+Here is how a ``_process()`` function might look like for you:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    func _process(delta):
+        if Engine.editor_hint:
+            # code to execute in editor
+        
+        if not Engine.editor_hint:
+            # code to execute in game
+        
+        # code to execute both in editor and in game
+
+.. note:: Modifications in editor are permanent. In our case, when we remove the script, the node will keep its rotation direction. Be careful so that you don't make unwanted modifications.
+
+Try it out
+-----------
+
+Add a ``Sprite`` node to your scene and set the texture to Godot icon. Attach and open a script, and change it to this:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    extends Sprite
+    tool
+
+    func _process(delta):
+        rotation_degrees += 180 * delta
+
+Save the script and return to the editor. You should now see your object rotate. If you run the game, it will also rotate.
+
+.. figure:: img/rotating-in_editor.gif
+
+.. note:: If you don't see the changes, reload the scene (close it and open it again).
+
+Now let's choose which code runs when. Modify your ``_process()`` function to look like this:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    func _process(delta):
+        if Engine.editor_hint:
+            rotation_degrees += 180 * delta
+        else:
+            rotation_degrees -= 180 * delta
+
+Save the script. Now the object will spin clockwise in the editor, but if you run the game, it will spin counter-clockwise.
+
+.. note:: Codes from other nodes don't run in the editor. Your access to other nodes is limited. You can access the tree and nodes, and their default properties, but you can't access user variables. If you want to do so, other nodes have to run in the editor too. AutoLoad nodes cannot be accessed in the editor at all.

```

Comment:
*Code from other nodes doesn't run in the editor.* (I'd avoid the plural "codes", sounds weird IMO)